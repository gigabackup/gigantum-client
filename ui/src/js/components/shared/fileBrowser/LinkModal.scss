@import '~Styles/_imports.scss';
.LinkModal{
  height: 100%;
}

.LinkModal__flex{
  height: 100%;
}
.LinkModal__container {
  @include flex(flex-start, row);
  flex-wrap: wrap;
  overflow: auto;
  height: 600px;
  border-bottom: 1px solid $slateBlue;
}

.Link__buttonContainer {
  margin-top: 10px;
  align-self: flex-end;
  width: 300px;
  @include flex(space-between, row);
}

.LinkModal__card{
  height: 260px;
  overflow: visible;
  &--border {
    border: 3px solid $slateBlue;
  }

  .Card{
    height: 230px;
  }
}

.LinkModal__dataset-container{
  margin-top: 100px;
  height: 376px;
  overflow: auto;
  width: 100%;
}

.LinkModal__wrapper{
  margin: 0 7px;
}
