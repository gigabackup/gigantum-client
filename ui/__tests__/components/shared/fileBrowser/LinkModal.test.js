// vendor
import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import relayTestingUtils from '@gigantum/relay-testing-utils';
// components
import LinkModal from 'Components/shared/fileBrowser/LinkModal';
import ReactDOM from 'react-dom';

ReactDOM.createPortal = node => node


const fixtures = {
  closeLinkModal: jest.fn(),
}

test('Test LinkModal', () => {
  const wrapper = renderer.create(
     <LinkModal {...fixtures}/>
  );
  const tree = wrapper.toJSON();
  expect(tree).toMatchSnapshot();
});
