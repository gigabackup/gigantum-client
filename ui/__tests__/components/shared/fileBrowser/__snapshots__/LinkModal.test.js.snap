// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test LinkModal 1`] = `
<div
  className="Modal"
>
  <div
    className="Modal__cover"
    onClick={[Function]}
  />
  <div
    className="Modal__content large"
    style={
      Object {
        "overflow": "hidden",
      }
    }
  >
    <div
      className="Modal__close"
      onClick={[Function]}
    />
    <div
      className="Modal__container"
    >
      <h4
        className="Modal__header"
      >
        Link Dataset
      </h4>
      <hr />
      <div
        className="Modal__sub-container"
      >
        <div
          className="LinkModal"
        >
          <div
            className="Loader"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
